TauAnalysis/MCEmbeddingTools/plugins/TrackMergeremb.cc

```diff
@@ -0,0 +1,391 @@
+    
+    edm::Handle<TrackToTrackMapnew> track_ref_map;
+    iEvent.getByToken(inputs_fixtrackrefs_, track_ref_map);
+    
+    edm::Handle<reco::TrackCollection> track_new_col;
+    iEvent.getByToken(inputs_fixtrackcol_, track_new_col);   
+    std::map<reco::TrackRef , reco::TrackRef > simple_track_to_track_map; //I didn't find a more elegant way, so just build a good old fassion std::map
+     for (unsigned abc =0;  abc < track_new_col->size();  ++abc) {
+      reco::TrackRef trackRef(track_new_col, abc);      
+      simple_track_to_track_map[((*track_ref_map)[trackRef])[0]] = trackRef;
+    }
+    
+    std::vector<reco::MuonRef> muonRefColl;
+    reco::MuonRefProd outputMuonsRefProd = iEvent.getRefBeforePut<reco::MuonCollection>();
+    unsigned new_idx=0;
+    for (auto akt_collection : to_merge){
+        edm::Handle<reco::MuonCollection> track_col_in;  
+        iEvent.getByToken(akt_collection, track_col_in);
+	unsigned old_idx=0;
+        for (reco::MuonCollection::const_iterator it = track_col_in->begin(); it != track_col_in->end(); ++it, ++old_idx, ++new_idx) {
+        outTracks->push_back( reco::Muon( *it ) );
+	 reco::MuonRef muRefold(track_col_in, old_idx);
+	 muonRefColl.push_back(muRefold);
+	 reco::MuonRef muRefnew(outputMuonsRefProd, new_idx);
+                  
+        if (it->track().isNonnull()){
+            //std::cout<<"pfmerge tr: "<<it->trackRef().id()<< " "<< it->trackRef().key()<< " " << simple_track_to_track_map[it->trackRef()].id() <<  " " << simple_track_to_track_map[it->trackRef()].key() <<std::endl;
+	    outTracks->back().setTrack( simple_track_to_track_map[it->track()]  );
+	 }
+        }
+	          
+    }// end merge 
+    
+     const int nMuons=outTracks->size();
+
+     std::vector<reco::MuonTimeExtra> dtTimeColl(nMuons);
+     std::vector<reco::MuonTimeExtra> cscTimeColl(nMuons);
+     std::vector<reco::MuonTimeExtra> combinedTimeColl(nMuons);
+     std::vector<reco::IsoDeposit> trackDepColl(nMuons);
+     std::vector<reco::IsoDeposit> ecalDepColl(nMuons);
+     std::vector<reco::IsoDeposit> hcalDepColl(nMuons);
+     std::vector<reco::IsoDeposit> hoDepColl(nMuons);
+     std::vector<reco::IsoDeposit> jetDepColl(nMuons);
+          
+    edm::OrphanHandle<reco::MuonCollection> muonHandle = iEvent.put(std::move(outTracks));
+    
+    auto fillMap = [](auto refH, auto& vec, edm::Event& ev, const std::string& cAl = ""){
+     typedef  edm::ValueMap<typename std::decay<decltype(vec)>::type::value_type> MapType;
+     std::unique_ptr<MapType > oMap(new MapType());
+     {
+       typename MapType::Filler filler(*oMap);
+       filler.insert(refH, vec.begin(), vec.end());
+       vec.clear();
+       filler.fill();
+     }
+     ev.put(std::move(oMap), cAl);
+    };
+    
+   fillMap(muonHandle, combinedTimeColl, iEvent, "combined");
+   fillMap(muonHandle, dtTimeColl, iEvent, "dt");
```

Comment:
is it meaningful to create emtpy collections.
Wouldn't it be easier to instruct the code consuming these to do nothing?